src/cpp/shared/scripting/callback.h

```diff
@@ -43,4 +51,26 @@ void ScriptCallback(const char * funcName, Args... args) {
   }
 }
 
+template<typename Res, typename... Args>
+nonstd::optional<Res> ScriptCallbackResult(const char * funcName, Args... args) {
+  if (!map_lua) { // if it's not an ironfist map
+    return nonstd::optional<Res>();
+  }
+
+  if (!LuaGlobalExists(map_lua, funcName)) {
+    return nonstd::optional<Res>();
+  }
+
+  lua_getglobal(map_lua, funcName);
+  ironfist_lua_pushmulti(args...);
+  int size = sizeof...(Args);
+  if (lua_pcall(map_lua, size, 1, 0) != LUA_OK) {
+    lua_pop(map_lua, 1); // Unsure if should have this
```

Comment:
Waiting on an answer to https://stackoverflow.com/questions/51031385/if-lua-pcall-returns-error-are-there-still-return-values-left-on-the-stack

From a quick glance at the implementation of Lua pcall, I believe this is not needed. However, I don't know enough about the Lua VM to know what exact behavior to look for if this is true vs. false.